spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,370 @@
+require "spec"
+require "yaml"
+
+private struct ObjectStub
+  getter called : Bool
+
+  def self.new(pull : YAML::PullParser)
+    pull.read_scalar
+    new(true)
+  end
+
+  def initialize(@called)
+  end
+end
+
+describe "Object.from_yaml" do
+  it "should create call .new on with a YAML::PullParser" do
+    ObjectStub.from_yaml("hello").called.should be_true
+  end
+end
+
+describe "Nil.from_yaml" do
+  it "should serialize and deserialize" do
+    Nil.from_yaml(nil.to_yaml).should eq nil
+  end
+
+  values = {"", "NULL", "Null", "null", "~"}
+
+  it "should return nil if a YAML null value" do
+    values.each do |value|
```

Comment:
Is it better to write? 

```crystal
  {"", "NULL", "Null", "null", "~"}.each do |value|
    it "should return nil if a YAML \"#{value}\" value" do
      # Test will an array since a standalone empty value shows as STREAM_END
      Array(Nil).from_yaml("- " + value).should eq [nil]
    end

    it "should raise if a YAML \"#{value}\" value is quoted with double quotes" do
      expect_raises(YAML::ParseException) do
        Nil.from_yaml %("#{value}")
      end
    end

    it "should raise if YAML \"#{value}\" value is quoted with single quotes" do
      expect_raises(YAML::ParseException) do
        Nil.from_yaml "'#{value}'"
      end
    end
  end
```

and so on

Reply:
Again... I did have this way but @straight-shoota suggested to inline all the tests into a single test. I would much rather have it the way above (the way I had it).

Reply:
@straight-shoota why? I prefer to know everything about broken spec. just by its description.

Reply:
As I've detailed above ([comment](https://github.com/crystal-lang/crystal/pull/4838#discussion_r134115351)), having a spec for each distinct value results in dozens of specs for actually the same behaviour. This is not necessary at all and results in slightly reduced test performance. We have fast-failing specs with multiple expectations all over stdlib. This might not always be a good idea but in this case I really see no benefit from having individual specs. If one of them fails, it's likely to assume others fail, too, and the reason is most likely the same. You'll have to run the tests again anyway. So we're not losing anything really but gaining execution speed for thousands of spec runs where there is no error in this part of the YAML parser.